application/src/main/java/org/togetherjava/tjbot/features/moderation/scam/TokenAnalyse.java

```diff
@@ -0,0 +1,157 @@
+package org.togetherjava.tjbot.features.moderation.scam;
+
+import org.togetherjava.tjbot.config.ScamBlockerConfig;
+import org.togetherjava.tjbot.features.utils.StringDistances;
+
+import java.net.URI;
+import java.util.Locale;
+
+/**
+ * Analyzes a given text token. Populates various metrics regarding the token possibly being
+ * suspicious, returning back results of the token analyze.
+ *
+ * Highly configurable, using {@link ScamBlockerConfig}. Entry point to use is
+ * {@link #analyze(String, AnalyseResults)}.
+ */
+final class TokenAnalyse {
+    private final ScamBlockerConfig config;
+
+    TokenAnalyse(ScamBlockerConfig config) {
+        this.config = config;
+    }
+
+    /**
+     * Analyzes the given token about being suspicious.
+     * 
+     * @param token the token to analyze
+     * @param results metrics representing how suspicious the token is
+     */
+    void analyze(String token, AnalyseResults results) {
+        if (token.isBlank()) {
+            return;
+        }
+
+        if (!results.pingsEveryone()
+                && ("@everyone".equalsIgnoreCase(token) || "@here".equalsIgnoreCase(token))) {
+            results.markPingsEveryone();
```

Comment:
These pre-checks really needed? They seem more like optimization, but would make the code a lot cleaner without them.